SDL_gpu_shadercross.h

```diff
@@ -38,23 +38,430 @@ SDL_GpuShaderFormat SDL_ShaderCross_GetShaderFormats(void)
+    0xB2,
+    0x20,
+    0x3F,
+    0x54
+};
+typedef struct IDxcCompiler3 IDxcCompiler3;
+typedef struct IDxcCompiler3Vtbl
+{
+    HRESULT(__stdcall *QueryInterface)(IDxcCompiler3 *This, REFIID riid, void **ppvObject);
+    ULONG(__stdcall *AddRef)(IDxcCompiler3 *This);
+    ULONG(__stdcall *Release)(IDxcCompiler3 *This);
+
+    HRESULT(__stdcall *Compile)(
+        IDxcCompiler3 *This,
+        const DxcBuffer *pSource,
+        LPCWSTR *pArguments,
+        Uint32 argCount,
+        void *pIncludeHandler, /* Technically IDxcIncludeHandler, but whatever... */
+        REFIID riid,
+        LPVOID *ppResult
+    );
+
+    HRESULT(__stdcall *Disassemble)(
+        IDxcCompiler3 *This,
+        const DxcBuffer *pObject,
+        REFIID riid,
+        LPVOID *ppResult
+    );
+} IDxcCompiler3Vtbl;
+struct IDxcCompiler3
+{
+    const IDxcCompiler3Vtbl *lpVtbl;
+};
+
+/* DXCompiler */
+static void *dxcompiler_dll = NULL;
+
+typedef HRESULT (__stdcall *DxcCreateInstanceProc)(
+    REFCLSID rclsid,
+    REFIID riid,
+    LPVOID *ppv);
+
+static DxcCreateInstanceProc SDL_DxcCreateInstance = NULL;
+static IDxcCompiler3 *SDL_DxcInstance = NULL;
+
+static void *SDL_ShaderCross_INTERNAL_CompileDXC(
+    SDL_GpuDevice *device,
+    void *createInfo,
+    const char *hlslSource,
+    const char *shaderProfile,
+    SDL_bool spirv)
+{
+    DxcBuffer source;
+    IDxcResult *dxcResult;
+    DXC_OUT_KIND blobOut;
+    IDxcBlob *blob;
+    IDxcBlobUtf8 *errors;
+    LPCWSTR args[] = {
+        L"-E",
+        L"main", /* FIXME */
+        NULL,
+        NULL,
+        NULL
+    };
+    Uint32 argCount = 2;
+    HRESULT ret;
+    void *result;
+
+    /* FIXME: dxcompiler could probably be loaded in a better spot */
+    if (dxcompiler_dll == NULL) {
+        dxcompiler_dll = SDL_LoadObject(DXCOMPILER_DLL);
+        if (dxcompiler_dll == NULL) {
+            return NULL;
+        }
+    }
+
+    if (SDL_DxcCreateInstance == NULL) {
+        SDL_DxcCreateInstance = (DxcCreateInstanceProc)SDL_LoadFunction(dxcompiler_dll, "DxcCreateInstance");
+        if (SDL_DxcCreateInstance == NULL) {
+            return NULL;
+        }
+    }
+
+    if (SDL_DxcInstance == NULL) {
+        SDL_DxcCreateInstance(&CLSID_DxcCompiler,
+                              IID_IDxcCompiler3,
+                              (void**) &SDL_DxcInstance);
+        if (SDL_DxcInstance == NULL) {
+            return NULL;
+        }
+    }
+
+    source.Ptr = hlslSource;
+    source.Size = SDL_strlen(hlslSource) + 1;
+    source.Encoding = 0; /* FIXME: The docs for this are a _bit_ scarce */
+
+    if (SDL_strcmp(shaderProfile, "ps_6_0") == 0) {
+        args[argCount++] = L"-T";
+        args[argCount++] = L"ps_6_0";
+    } else if (SDL_strcmp(shaderProfile, "vs_6_0") == 0) {
+        args[argCount++] = L"-T";
+        args[argCount++] = L"vs_6_0";
+    } else if (SDL_strcmp(shaderProfile, "cs_6_0") == 0) {
+        args[argCount++] = L"-T";
+        args[argCount++] = L"cs_6_0";
+    }
```

Comment:
Why not just forward this directly? SDL has `SDL_iconv_string()` for the encoding conversion. There are a few convenient macros around it in `src/core/windows/SDL_windows.h` that are used for windows things, though they're not public API.